source/reference/mongotranslate.txt

```diff
@@ -0,0 +1,123 @@
+==================
+``mongotranslate``
+==================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+.. only:: html
+
+   .. meta::
+      :description: The mongotranslate reference page.
+      :keywords: mongotranslate, mongodb, man page
+
+Description
+-----------
+
+.. only:: (not man)
+
+   .. binary:: mongotranslate
+
+``mongotranslate`` takes a SQL query and returns
+an equivalent MongoDB :manual:`aggregation pipeline </aggregation>`. It
+is designed to help users understand how |bi-short| translates
+SQL queries for use with MongoDB databases.
+
+``mongotranslate`` is a standalone program. It requires
+a :doc:`drdl </reference/mongodrdl>` file generated by
+:binary:`~bin.mongodrdl`, but does not require
+:binary:`~bin.mongosqld` or :binary:`~bin.mongod` to be running at the
+time of execution.
+
+Usage
+-----
+
+``mongotranslate`` has the following syntax:
+
+.. code-block:: javascript
+
+   mongotranslate [--query | --queryFile] <query | queryfile> [options]
+
+Command Line Options
+--------------------
+
+.. program:: mongotranslate
+
+.. include:: /includes/option/option-mongotranslate-query.rst
+
+.. include:: /includes/option/option-mongotranslate-queryFile.rst
+
+.. include:: /includes/option/option-mongotranslate-schema.rst
+
+.. include:: /includes/option/option-mongotranslate-dbName.rst
+
+.. include:: /includes/option/option-mongotranslate-format.rst
+
+.. include:: /includes/option/option-mongotranslate-explain.rst
+
+.. include:: /includes/option/option-mongotranslate-mongoversion.rst
+
+Examples
+--------
+
+The following example specifies an inline query for translation:
+
+.. code-block:: javascript
+
+   mongotranslate --query=“select test.name from restaurants where name like 'Brooklyn%'” \
+   --schema=schema.drdl
+
+The above command returns the following results:
+
+.. code-block:: json
+   :copyable: false
+
+   [
+     {"$match": {"name": {"$regex": "^Brooklyn.*$","$options": "i"}}},
+     {"$project": {"test_DOT_restaurants_DOT_name": "$name","_id": {"$numberInt":"0"}}},
+   ]
+
+.. note::
+
+   If you need to quote a string inside the :option:`--query
+   <mongotranslate --query>` parameter, be sure to use single quotes.
```

Comment:
I really like having this note here! :) We also ran into the issue of customers using back-ticks in their queries, which are escaped by Bash. We added the `queryFile` option for this reason, so we could also add: `If you must use a double quote or back-tick in your query, pass in your query using the --queryFile option.`

Reply:
Done.